components/common/confirmPurchaseModal/ConfirmPurchaseModal.vue

```diff
@@ -104,7 +105,14 @@ const isDisabledPurchasePrefix = isRemark
 const autoteleport = ref()
 const actions = computed(() => [props.action])
 
-const loading = computed(() => !autoteleport.value?.isReady || props.loading)
+const loading = ref(props.loading)
+
+watch(
+  () => autoteleport.value?.isReady,
+  (val) => {
+    loading.value = !val
+  },
+)
```

Comment:
@daiagi no changes needed here 

this is the initial code 

![CleanShot 2024-02-21 at 14 57 27@2x](https://github.com/kodadot/nft-gallery/assets/44554284/ca758269-ced3-4b4f-bcfc-72cc1ad2ed33)

I think it's does the same as your version 

the issue https://github.com/kodadot/nft-gallery/issues/9477 got already fixed by  #9491

https://github.com/kodadot/nft-gallery/pull/9484#issuecomment-1954109180


if anything, we can handle the error case inside the subscription 

![CleanShot 2024-02-21 at 14 55 35@2x](https://github.com/kodadot/nft-gallery/assets/44554284/b43e0b31-841c-42b5-ad43-d367636760c7)